sharktank/sharktank/models/llama/testing.py

```diff
@@ -163,3 +164,21 @@ def make_random_llama_theta(
     )
 
     return Theta(res)
+
+
+def make_random_theta_from_dataset(dataset: Dataset) -> Theta:
```

Comment:
This function seems general enough to go into
`sharktank/sharktank/layers/testing.py`
or into a new file
`sharktank/sharktank/types/testing.py`

It would also benefit from a short description. You could just move comment form below.

Also from a symmetry view point it should accept a `Theta` and return a `Theta`.

You could have two function `make_random_theta_from_theta` and `make_random_dataset_from_dataset`.
`make_random_dataset_from_dataset` would just additionally copy the properties.